aten/src/ATen/native/cpu/ScatterGatherKernel.cpp

```diff
@@ -275,7 +297,158 @@ void scatter_add_cpu_kernel(Tensor& self, int64_t dim, const Tensor& index, cons
         self_data[idx_dim * self_dim_stride] += src_data[i * src_dim_stride];
       }
     },
-      /*serial_exec=*/true);
+    /*serial_exec=*/false);
+}
+
+  class ReduceFunctor {
+  public:
+    ReduceFunctor() {};
+    template <typename T, typename func_t>
+    void operator() (T* self_data, const T* src_data,
+                     func_t op) {
+      op(self_data, src_data);
+    }
+  };
+
+  ReduceFunctor reduce_fn;
+
+  auto reduce_sum = [](auto * self_data, const auto * src_data) {
+                      *self_data += *src_data;
+                    };
```

Comment:
Just curious. Why you use `T*` in one case and `auto *` in the other? I mean the space in-between.

Reply:
Also, remove `const` from `src_data`. This way we can pass the arguments in a fixed order inside the base kernel, and flip the order for `gather_` in a kernel callback.